### PR TITLE
Added tests for recipient controller

### PR DESCRIPTION
**Pull request by joffe97**

### CODE REVIEW

File: GiEnJul/Controllers/RecipientController.cs [hunk 1]
```diff
@@ -26,6 +26,11 @@ public RecipientController(IRecipientRepository recipientRepository, IPersonRepo
         public async Task<ActionResult<Entities.Recipient>> PostAsync([FromBody] Models.Recipient recipient)
         {
             _log.Debug("Adding recipient object: {@recipient}", recipient);
+            if (string.IsNullOrEmpty(recipient.Location)) {
+                var e = new Exception("Cannot add recipient with no location.");
+                _log.Error("Exception occurred while trying to add recipient: {@0}. \n{@1}", recipient, e);
+                throw e;
```
Comment (RonHaland) [inline GiEnJul/Controllers/RecipientController.cs:32]:
```suggestion
                _log.Error("Exception occurred while trying to add recipient: {@0}", recipient);
                throw new ArgumentNullException("Location", "Cannot add recipient with no location");
```

Not sure if I used everything correctly here, maybe test this locally first

**Review by RonHaland — DISMISSED** [review]

File: GiEnJul/Controllers/RecipientController.cs [hunk 2]
```diff
@@ -26,6 +26,11 @@ public RecipientController(IRecipientRepository recipientRepository, IPersonRepo
         public async Task<ActionResult<Entities.Recipient>> PostAsync([FromBody] Models.Recipient recipient)
         {
             _log.Debug("Adding recipient object: {@recipient}", recipient);
+            if (string.IsNullOrEmpty(recipient.Location)) {
```
Comment (Sommerli) [inline GiEnJul/Controllers/RecipientController.cs:29]: `if (string.IsNullOrEmpty(recipient.Location))`

This behaviour could be handled using Annotations in the Models.Recipient Class. 
Example
`[Required]
public string Location { get; set; }
`

Using Annotations, the method will return an error if Location is not set.

File: GiEnJul/Controllers/RecipientController.cs (hunk 1 above)
Comment (joffe97) [reply GiEnJul/Controllers/RecipientController.cs]: Moved the null value check into the controller.

**Comment by Sommerli** [comment]
I belive some changes will have to be made, with Issue #135. 

I am not sure if this PR should be "on-hold" until 135 is resolved.

**Review by Sommerli — APPROVED** [review]

**Pull request merged**